File: DSA/fundamentals/maps/part0/javascript/judge.js
const Solution = require('./solution');

class Judge {
    runTests() {
        const testCasesN = [
            ["Doe, John", "Smith, Jane", "Johnson, Jim"],
            ["Doe, John", "Smith, Jane", "Johnson, Jim", "Williams, Mary", "Brown, Bob"],
        ];

        let passes = 0;
        for (let i = 0; i < testCasesN.length; i++) {
            if (this.test(testCasesN[i], this.testCaseSolutionOutput(testCasesN[i]))) {
                passes++;
            }
        }
        console.log(`Passed ${passes} of ${testCasesN.length} tests.`);
    }

    test(n, expected) {
        const solution = new Solution();
        const userMap = new Map();
        solution.storeNames(n, userMap);
        
        const userValues = Array.from(userMap.values()).sort();
        const expectedValues = Array.from(expected.values()).sort();
        const passed = userMap.size === expected.size && 
                      JSON.stringify(userValues) === JSON.stringify(expectedValues);

        console.log(`n: ${JSON.stringify(n)}`);
        console.log(`Expected Output: ${JSON.stringify(Array.from(expected.values()))}`);
        console.log(`Your Output: ${JSON.stringify(Array.from(userMap.values()))}`);
        console.log(passed ? "Test PASSED" : "Test FAILED");
        console.log("-----------");
        return passed;
    }

    testCaseSolutionOutput(n) {
        const expectedMap = new Map();
        for (let i = 0; i < n.length; i++) {
            const parts = n[i].split(", ");
            expectedMap.set(i.toString(), parts[1] + " " + parts[0]);
        }
        return expectedMap;
    }
}

module.exports = Judge;

File: DSA/fundamentals/maps/part0/python/judge.py
from solution import Solution

class Judge:
    def run_tests(self):
        test_cases_n = [
            ["Doe, John", "Smith, Jane", "Johnson, Jim"],
            ["Doe, John", "Smith, Jane", "Johnson, Jim", "Williams, Mary", "Brown, Bob"],
        ]

        passes = 0;
        for i in range(len(test_cases_n)):
            if self.test(test_cases_n[i].copy(), self.test_case_solution_output(test_cases_n[i].copy())):
                passes += 1
        print(f"Passed {passes} of {len(test_cases_n)} tests.")

    def test(self, n, expected):
        solution = Solution()
        user_map = {}
        solution.store_names(n, user_map)
        passed = len(user_map) == len(expected) and set(user_map.values()) == set(expected.values())

        print(f"n: {n}")
        print(f"Expected Output: {list(expected.values())}")
        print(f"Your Output: {list(user_map.values())}")
        print("Test " + ("PASSED" if passed else "FAILED"))
        print("-----------")
        return passed

    def test_case_solution_output(self, n):
        expected_map = {}
        for i, name in enumerate(n):
            parts = name.split(", ")
            expected_map[str(i)] = parts[1] + " " + parts[0]
        return expected_map